plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/ModulesManager.java

```diff
@@ -1171,6 +1177,29 @@ public AdditionalGrammarVersionsToCheck getAdditionalGrammarVersions()
 
                     }
                 }
+            } else if ("typing".equals(n.getName())) {
+                Module module = (Module) ((SourceModule) n).getAst();
+                for (SimpleNode node : module.body) {
+                    if (node instanceof Assign && ((Assign) node).value instanceof Call
+                            && ((Call) ((Assign) node).value).func instanceof Name
+                            && ((Name) ((Call) ((Assign) node).value).func).id.equals("_alias")) {
+                        Assign assign = (Assign) node;
+                        exprType[] args = ((Call) assign.value).args;
+                        sliceType[] dims = null;
+                        if (args[1] instanceof org.python.pydev.parser.jython.ast.Tuple) {
```

Comment:
This seems a bit odd for me, why are you getting `args[1]`? 

What you want is convert:
`Dict = _alias(dict, (KT, VT), inst=False)`


to:

`Dict = dict`

and what you seem to be doing is converting

`Dict = _alias(dict, (KT, VT), inst=False)`

to `Dict = [KV, VT]`

p.s.: so, just check if args[0] is a name or attribute and set it to the value.

Reply:
The args[0] is already being defined to value at

https://github.com/fabioz/Pydev/blob/97b26303f141b242c577ccd39787017ae9b9cdbe/plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/ModulesManager.java#L1200

And I did not check if it is a name or attribute because Subscript conscructor receives an exprType as value (and args[0] will always be an exprType in this 'if' scope). So I found no reason to do an handle like this

Reply:
I think that the only thing you need to do for PyDev right now is `assign.value = args[0]` (I believe PyDev does not use the Subscript info provided at this place for type inference, so, it just complicates things a bit further on).